hooks/build

```
@@ -0,0 +1,449 @@
+# It will also help with simpler skipping of unused images.
+export DOCKER_BUILDKIT=${DOCKER_BUILDKIT:=0}
+
+# List of images to push at the end of the build
+IMAGES_BUILT=""
+
+end_step
+
+AIRFLOW_CONTAINER_CLEANUP_IMAGES=${AIRFLOW_CONTAINER_CLEANUP_IMAGES:="false"}
+if [[ "${AIRFLOW_CONTAINER_CLEANUP_IMAGES}" == "true" ]]; then
+    echo
+    ${AIRFLOW_ROOT}/confirm "Removing all local images for that build."
+    echo
+    start_step "Removing images"
+    docker rmi ${PYTHON_BASE_IMAGE} || true
+    docker rmi ${AIRFLOW_IMAGE} || true
+    docker rmi ${AIRFLOW_CI_IMAGE} || true
+    echo
+    echo "###################################################################"
+    echo "NOTE!! Removed Airflow images for Python version ${PYTHON_VERSION}."
+    echo "       But the disk space in docker will be reclaimed only after"
+    echo "       running 'docker system prune' command."
+    echo "###################################################################"
+    echo
+    end_step
+    exit 0
+fi
+
+start_step "Pulling images to populate cache"
+
+if [[ "${AIRFLOW_CONTAINER_USE_PULLED_IMAGES_CACHE}" == "true" ]]; then
+    echo
+    echo "Pulling images"
+    echo
+    echo
+    if [[ "${AIRFLOW_CONTAINER_FORCE_PULL_IMAGES}" == "true" ]]; then
+        echo "Force-pull base python image. This forces to get the latest version."
+        echo
+        set -x
+        docker pull ${PYTHON_BASE_IMAGE}
+        set +x
+        echo
+    fi
+    IMAGES_TO_PULL="${AIRFLOW_CI_IMAGE}"
+    if [[ "${AIRFLOW_CONTAINER_SKIP_MAIN_IMAGE}" == "true" ]]; then
+        echo "Skip downloading the main Airflow image"
+    else
+        IMAGES_TO_PULL="${IMAGES_TO_PULL} ${AIRFLOW_IMAGE}"
+    fi
+
+    DOCKER_CACHE_DIRECTIVE=""
+    for IMAGE in ${IMAGES_TO_PULL}
+    do
+        echo
+        echo "Checking whether image ${IMAGE} needs to be pulled."
+        echo
+        PULL_IMAGE="false"
+        if [[ "${AIRFLOW_CONTAINER_FORCE_PULL_IMAGES}" == "true" ]]; then
+            echo
+            echo "Pulling images is forced. Pulling ${IMAGE}"
+            echo
+            PULL_IMAGE="true"
+        else
+            IMAGE_HASH=$(docker images -q ${IMAGE} 2> /dev/null)
+            if [[ "${IMAGE_HASH}" == "" ]]; then
+                echo
+                echo "No image ${IMAGE} locally available. Pulling for the first time."
+                echo
+                PULL_IMAGE="true"
+            else
+                echo
+                echo "Image ${IMAGE} is in local registry (${IMAGE_HASH}). Not pulling it!"
+                echo
+                PULL_IMAGE="false"
+            fi
+        fi
+        if [[ "${PULL_IMAGE}" == "true" ]]; then
+            if [[ "${AIRFLOW_CONTAINER_SKIP_PULLING_AIRFLOW_IMAGES}" == "true" ]]; then
+                echo
+                echo "Skipping pulling image ${IMAGE}"
+                echo
+            else
+                echo
+                set -x
+                docker pull ${IMAGE} || true
```

```suggestion
                docker pull "${IMAGE}" || true
```
Double quote to prevent globbing and word splitting.

---

Done